### PR TITLE
Added functionality to make screenshot on test failure

### PR DESCRIPTION
**Pull request by Nikodych**
Changed base class name from "Test" into "TestClass".
Created new class "TestListener" with functionality to make screenshot on test failure, by overriding testng listeners and function "takeScreenShotOnFailure".
In "TestClass" created an example test to verify work of "TestListener".
Screenshot file is located at "oms_test_framework/screenshot/".
Listeners “@Listeners ({TestListener.class})” added at class level. When run above test file. You should see that screen shot will capture and save into screenshot folder with same name as test name.

### CODE REVIEW

File: src/main/java/academy/softserve/edu/TestListener.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+public class TestListener extends TestListenerAdapter {
+
+    @Override
+    public void onTestFailure(final ITestResult iTestResult) {
+        takeScreenShotOnFailure(iTestResult);
+        System.out.println("TestClass has failed: " + iTestResult.getName());
+    }
+
+    @Override
```
Comment (toleksyn) [inline src/main/java/academy/softserve/edu/TestListener.java]: Remove this method

File: src/main/java/academy/softserve/edu/TestListener.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public void onTestSuccess(final ITestResult iTestResult) {
+        System.out.println("Test Passed: " + iTestResult.getName());
+    }
+
+    @Override
```
Comment (toleksyn) [inline src/main/java/academy/softserve/edu/TestListener.java]: Remove this method

File: src/main/java/academy/softserve/edu/TestListener.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    public void onTestSkipped(final ITestResult iTestResult) {
+        System.out.println("Test Skipped: " + iTestResult.getName());
+    }
+
+    public void takeScreenShotOnFailure(final ITestResult iTestResult) {
+        final Object object = iTestResult.getInstance();
+        final WebDriver driver = ((TestClass) object).getDriver();
+
```
Comment (toleksyn) [inline src/main/java/academy/softserve/edu/TestListener.java]: STICK - what should be here?

File: src/main/java/academy/softserve/edu/Test.java [hunk 4]
```diff
@@ -1,4 +0,0 @@
-package academy.softserve.edu;
```
Comment (toleksyn) [inline src/main/java/academy/softserve/edu/Test.java]: Remove screenshot file

File: src/main/java/academy/softserve/edu/TestListener.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import java.io.File;
+import java.io.IOException;
+
+public class TestListener extends TestListenerAdapter {
+
+    @Override
+    public void onTestFailure(final ITestResult iTestResult) {
+        takeScreenShotOnFailure(iTestResult);
+        System.out.println("TestClass has failed: " + iTestResult.getName());
```
Comment (toleksyn) [inline src/main/java/academy/softserve/edu/TestListener.java]: remove println

File: src/main/java/academy/softserve/edu/TestListener.java (hunk 1 above)
Comment (Nikodych) [reply src/main/java/academy/softserve/edu/TestListener.java]: removed

File: src/main/java/academy/softserve/edu/TestListener.java (hunk 2 above)
Comment (Nikodych) [reply src/main/java/academy/softserve/edu/TestListener.java]: removed

File: src/main/java/academy/softserve/edu/TestListener.java (hunk 3 above)
Comment (Nikodych) [reply src/main/java/academy/softserve/edu/TestListener.java]: absence "final" for "File" =(

File: src/main/java/academy/softserve/edu/Test.java (hunk 4 above)
Comment (Nikodych) [reply src/main/java/academy/softserve/edu/Test.java]: removed

File: src/main/java/academy/softserve/edu/TestListener.java (hunk 5 above)
Comment (Nikodych) [reply src/main/java/academy/softserve/edu/TestListener.java]: removed

File: src/test/java/academy/softserve/edu/utils/TestListener.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import java.io.IOException;
+
+public class TestListener extends TestListenerAdapter {
+
+    @Override
+    public void onTestFailure(final ITestResult iTestResult) {
+        takeScreenShotOnFailure(iTestResult);
+    }
+
+    public void takeScreenShotOnFailure(final ITestResult iTestResult) {
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/utils/TestListener.java]: why is this public?

File: src/test/java/academy/softserve/edu/utils/TestListener.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        final Object object = iTestResult.getInstance();
+        final WebDriver driver = ((TestClassForListener) object).getDriver();
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/utils/TestListener.java]: driver = ((TestClassForListener) object).getDriver(); - will only work for TestClassForListener.class, make it work fer every class

File: src/test/java/academy/softserve/edu/utils/TestListener.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        final File screenShotFile = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
+        try {
+            FileUtils.copyFile(screenShotFile, new File("screenshot/" + iTestResult.getName() + ".png"));
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/utils/TestListener.java]: make screenshot name include current date in format:
2016-09-07 - <testMethodName>.png

File: src/test/java/academy/softserve/edu/utils/TestListener.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Override
+    public void onTestFailure(final ITestResult iTestResult) {
+        takeScreenShotOnFailure(iTestResult);
+    }
+
+    public void takeScreenShotOnFailure(final ITestResult iTestResult) {
+        final Object object = iTestResult.getInstance();
+        final WebDriver driver = ((TestClassForListener) object).getDriver();
+        final File screenShotFile = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
```
Comment (toleksyn) [inline src/test/java/academy/softserve/edu/utils/TestListener.java:29]: Save image file in $project_dir/test-output/html/screenshots/

File: src/test/java/academy/softserve/edu/utils/TestListener.java (hunk 6 above)
Comment (Nikodych) [reply src/test/java/academy/softserve/edu/utils/TestListener.java]: done

File: src/test/java/academy/softserve/edu/utils/TestListener.java (hunk 8 above)
Comment (Nikodych) [reply src/test/java/academy/softserve/edu/utils/TestListener.java]: done

File: src/test/java/academy/softserve/edu/utils/TestListener.java (hunk 9 above)
Comment (Nikodych) [reply src/test/java/academy/softserve/edu/utils/TestListener.java:29]: done

**Pull request merged**